daemon/errors.go

```diff
@@ -1,53 +1,112 @@
 package daemon
 
 import (
-	"fmt"
-
-	"github.com/docker/docker/api/errors"
+	"github.com/pkg/errors"
 )
 
-func (d *Daemon) imageNotExistToErrcode(err error) error {
-	if dne, isDNE := err.(ErrImageDoesNotExist); isDNE {
-		return errors.NewRequestNotFoundError(dne)
-	}
-	return err
+func errNotRunning(id string) error {
+	return stateConflictError{errors.Errorf("Container %s is not running", id)}
 }
 
-type errNotRunning struct {
-	containerID string
+type notFoundError struct {
```

Comment:
If it doesn't add much more code, it might be cleaner to have specific error types for containers, volumes, images, etc. which are not found, with the errors generated in the `Error` method instead of returning text wrapped by the type.